client/source/client.cpp

```diff
@@ -0,0 +1,610 @@
+
+    if (inet_pton(AF_INET, serverAddress.c_str(), &address.sin_addr) <= 0)
+    {
+        perror("Invalid address/Address not supported");
+        exit(EXIT_FAILURE);
+    }
+
+    if (connect(clientSocket, (struct sockaddr *)&address, sizeof(address)) < 0)
+    {
+        perror("Connection failed");
+        exit(EXIT_FAILURE);
+    }
+}
+
+void Client::sendFeedback()
+{
+    int foodItemId;
+    int rating;
+    std::string employeeId;
+    std::string comment;
+
+    std::cout << "Enter Food Item ID: ";
+    std::cin >> foodItemId;
+    std::cin.ignore();
+    std::cout << "Enter Employee ID: ";
+    std::getline(std::cin, employeeId);
+
+    std::cout << "Enter Rating(0-5): ";
+    std::cin >> rating;
+    std::cin.ignore();
+
+    std::cout << "Enter Comment: ";
+    std::getline(std::cin, comment);
+    std::string message = "ADD_FEEDBACK:" + employeeId + ":" + std::to_string(foodItemId) + ":" + std::to_string(rating) + ":" + comment;
+    send(clientSocket, message.c_str(), message.length(), 0);
+    std::cout << "Feedback sent\n";
+
+    char buffer[bufferSize] = {0};
+    read(clientSocket, buffer, bufferSize);
+    std::cout << "Response from server: " << buffer << std::endl;
+}
+void Client::viewNotificationsRequestToServer(const std::string &message)
+{
+    send(clientSocket, message.c_str(), message.length(), 0);
+    std::cout << "Notifications request sent\n";
+
+    char buffer[bufferSize] = {0};
+    read(clientSocket, buffer, bufferSize);
+    std::cout << "Notifications:\n" << buffer << std::endl;
+}
+
+void Client::getRecommendations(std::string message)
+{
+    send(clientSocket, message.c_str(), message.length(), 0);
+    std::cout << "Recommendation request sent\n";
+
+    char buffer[bufferSize] = {0};
+    read(clientSocket, buffer, bufferSize);
+
+    std::string response(buffer);
+
+    std::istringstream recommendationsStream(response);
+    std::string recommendation;
+    std::vector<std::string> recommendations;
+    while (std::getline(recommendationsStream, recommendation, '|'))
+    {
+        recommendations.push_back(recommendation);
+    }
+
+    std::cout << "Recommendations received:\n";
+    for (const auto &rec : recommendations)
+    {
+        std::istringstream recommendationStream(rec);
+        std::string token;
+        std::vector<std::string> fields;
+        while (std::getline(recommendationStream, token, ';'))
+        {
+            fields.push_back(token);
+        }
+
+        if (fields.size() == 5)
+        {
+            std::cout << "Food Item: " << fields[1] << "\n";
+            std::cout << "Price: " << fields[2] << "\n";
+            std::cout << "Rating: " << fields[4] << "\n";
+            std::cout << "Availability: " << (fields[3] == "1" ? "Yes" : "No") << "\n";
+            std::cout << "------------------\n";
+        }
+    }
+
+    storeRecommendations(recommendations);
+}
+
+void Client::storeRecommendations(std::vector<std::string> recommendations)
```

Comment:
Why are all these methods in the client class itself? Can we have separate classes created to handle SRP and have only relevant code in them.